src/convex/ConvexAvatarMultiToken.sol

```diff
@@ -0,0 +1,698 @@
+        baseRewardPools.remove(crvRewards);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // PUBLIC: Owner/Manager
+    ////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Claim and process CRV, CVX & FXS rewards, selling them for stables: USDC & FRAX
+    /// @dev This can be called by the owner or manager to opportunistically harvest in good market conditions.
+    /// @return processed_ An array containing addresses and amounts of harvested tokens (i.e. tokens that have finally
+    ///                    been swapped into).
+    function processRewards() external onlyOwnerOrManager returns (TokenAmount[] memory processed_) {
+        processed_ = _processRewards();
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // PUBLIC: Keeper
+    ////////////////////////////////////////////////////////////////////////////
+
+    /// @notice A function to process pending CRV, CVX & FXS rewards at regular intervals. Can only be called by the
+    ///         keeper when the contract is not paused.
+    /// @param _performData The calldata to be passed to the upkeep function. Not used!
+    function performUpkeep(bytes calldata _performData) external override onlyKeeper whenNotPaused {
+        _processRewardsKeeper();
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // INTERNAL
+    ////////////////////////////////////////////////////////////////////////////
+
+    function _createPrivateVault(uint256 _pid) internal {
+        (,, address stakingTokenAddr,, uint8 active) = CONVEX_FRAX_REGISTRY.poolInfo(_pid);
+        if (active == 0) {
+            revert PoolDeactivated(_pid);
+        }
+
+        address vaultAddr = FRAX_BOOSTER.createVault(_pid);
+        IERC20MetadataUpgradeable(stakingTokenAddr).safeApprove(vaultAddr, type(uint256).max);
+
+        // NOTE: we should store the `vaultAddr` on storage for ease of deposits/wds/reward claims
+        privateVaults[_pid] = vaultAddr;
+        pidsPrivateVaults.add(_pid);
+    }
+
+    function _addCurveLpPositionInfo(uint256 _newPid) internal {
+        pids.add(_newPid);
+        (address lpToken,,, address crvRewards,,) = CONVEX_BOOSTER.poolInfo(_newPid);
+        assets.add(lpToken);
+        baseRewardPools.add(crvRewards);
+        // NOTE: during new lp addition approve those assets to convex booster
+        IERC20MetadataUpgradeable(lpToken).safeApprove(address(CONVEX_BOOSTER), type(uint256).max);
+    }
+
+    /// @notice Unstakes the given amount of assets and transfers them back to owner.
+    /// @dev This function doesn't claim any rewards. Caller can only be owner.
+    /// @param _pids Pids to be targetted to unstake from.
+    /// @param _bptsDeposited Amount of assets to be unstaked.
+    function _withdraw(uint256[] memory _pids, uint256[] memory _bptsDeposited) internal {
```

Comment:
```suggestion
    function _withdraw(uint256[] calldata _pids, uint256[] calldata _bptsDeposited) internal {
```

Reply:
resolved in 8f1285e8799cd5cb413b8cb597f1829926afab18